Source/Particles/Filter/FilterFunctors.H

```diff
@@ -106,8 +106,13 @@ struct ParserFilter
     {
         if ( !m_is_active ) return 1;
         amrex::Real const x  = p.pos(0);
-        amrex::Real const y  = p.pos(1);
-        amrex::Real const z  = p.pos(2);
+#if AMREX_SPACEDIM == 3
+        auto const y  = p.pos(1);
+        auto const z  = p.pos(2);
+#else
+        amrex::ParticleReal const y  = 0.0;
+        auto const z  = p.pos(1);
+#endif
```

Comment:
We don't currently have a `GetParticlePosition` for `SuperParticleType`, so maybe the easiest thing to do here is add another branch to the `ifdef` for RZ?

Reply:
In RZ we also just copy the 0 and 1 position, IIRC: https://warpx.readthedocs.io/en/latest/developers/dimensionality.html